app/models/usage_update.rb

```diff
@@ -0,0 +1,213 @@
+    # until_date = options[:until_date].present? ? Date.parse(options[:until_date]) : Date.current
+
+    usage_update = UsageUpdate.new
+    # usage_update.queue_jobs(usage_update.unfreeze(from_date: from_date.strftime("%F"), until_date: until_date.strftime("%F")))
+    usage_update.queue_jobs 
+  end
+
+  def source_id
+    "usage_update"
+  end
+
+
+  def process_data options={} 
+    messages = get_query_url options
+    messages.each do |message|
+      body = JSON.parse(message.body)
+      report_id = body["report_id"]
+      UsageUpdateParseJob.perform_later(report_id, options)
+      delete_message message
+    end if messages.respond_to?("each")
+    messages.length
+  end
+
+  def get_query_url options={}
+    queue_url = sqs.get_queue_url(queue_name: "#{Rails.env}_usage" ).queue_url
+    resp = sqs.receive_message(queue_url: queue_url, max_number_of_messages: 5, wait_time_seconds: 1)
+    resp.messages
+  end
+
+  def self.get_data report_id, options={}
+    host = URI.parse(report_id).host.downcase
+    report = Maremma.get(report_id, timeout: 120, host: host)
+    report
+  end
+
+  def sqs
+    sqs = Aws::SQS::Client.new(region: ENV["AWS_REGION"])
+    puts sqs.get_queue_url(queue_name: "#{Rails.env}_usage" ).queue_url
+    sqs
+  end
+
+
+  def get_total options={}
+    queue_url = sqs.get_queue_url(queue_name: "#{Rails.env}_usage" ).queue_url
+    req = sqs.get_queue_attributes(
+      {
+        queue_url: queue_url, attribute_names: 
+          [
+            'ApproximateNumberOfMessages', 
+            'ApproximateNumberOfMessagesNotVisible'
+          ]
+      }
+    )
+
+    msgs_available = req.attributes['ApproximateNumberOfMessages']
+    msgs_available.to_i
+  end
+
+  def queue_jobs(options={})
+
+    total = get_total(options)
+    
+    if total < 1
+      text = "No works found for in the Usage Reports Queue."
+    end
+
+    num_messages = total
+    while num_messages > 0 
+        queued = process_data(options)
+        num_messages -= queued
+        puts num_messages
+        puts queued
+    end
+    text = "#{queued} reports queued out of #{total} for Usage Reports Queue"
+
+    Rails.logger.info text
+    # send slack notification
+    if queued == 0
+      options[:level] = "warning"
+    else
+      options[:level] = "good"
+    end
+    options[:title] = "Report for #{source_id}"
+    send_notification_to_slack(text, options) if options[:slack_webhook_url].present?
+    queued
+  end
+
+  def self.parse_data report, options={}
+
+    return report.body.fetch("errors") if report.body.fetch("errors", nil).present?
+    return [{ "errors" => { "title" => "The report is blank" }}] if report.body.blank?
+
+    items = report.body.dig("data","report","report-datasets")
+    header = report.body.dig("data","report","report-header")
+    report_id = report.url
+
+    created = header.fetch("created")
+    Array.wrap(items).reduce([]) do |x, item|
+      data = {}
```

Comment:
Please rewrite to `data = { doi: item.dig("dataset-id").first.dig("value"), pid: normalize_doi(data[:doi]) }`, etc.